### PR TITLE
[refactor-unit-tests] Various unit test refactor, and amended implici…

### PR DESCRIPTION
**Pull request by MJCallahanPage**
…t date formatter

### CODE REVIEW

File: app/utils/ImplicitDateFormatter.scala [hunk 1]
```diff
@@ -18,9 +18,20 @@ package utils
...
 import java.time.format.DateTimeFormatter
+import java.time.format.TextStyle._
+import java.util.Locale._
 
 trait ImplicitDateFormatter {
-  implicit val localDate: String => LocalDate = date => LocalDate.parse(date, DateTimeFormatter.ofPattern("uuuu-M-d"))
+
+  implicit def toLocalDate(s: String): LocalDate = localData(s).toLocalDate
+
+  implicit class localData(s: String) {
```
Comment (MarkAKelly) [inline app/utils/ImplicitDateFormatter.scala]: Is this a small typo on localDate, or is it meant to be localData?

File: conf/logback-test.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+
+    <appender name="FILE" class="ch.qos.logback.core.FileAppender">
+        <file>logs/income-tax-subscription.log</file>
```
Comment (MarkAKelly) [inline conf/logback-test.xml]: income-tax-view-change.log?

Comment (MJCallahanPage) [inline conf/logback-test.xml]: Whooops, that's what happens when you steal

File: app/utils/ImplicitDateFormatter.scala (hunk 1 above)
Comment (MJCallahanPage) [inline app/utils/ImplicitDateFormatter.scala]: Correct, should be localDate silly typo. Well spotted :+1:

**Pull request merged**